tests/chainer_tests/functions_tests/evaluation_tests/test_classification_summary.py

```diff
@@ -0,0 +1,134 @@
+    tp = numpy.zeros((label_num,), dtype=numpy.int32)
+    support = numpy.zeros((label_num,), dtype=numpy.int32)
+    for p, t in zip(preds.ravel(), ts.ravel()):
+        if t == ignore_label:
+            continue
+        support[t] += 1
+        if p == t:
+            tp[t] += 1
+    return dtype(tp) / support
+
+
+def precision(preds, ts, dtype, label_num, ignore_label):
+    tp = numpy.zeros((label_num,), dtype=numpy.int32)
+    relevant = numpy.zeros((label_num,), dtype=numpy.int32)
+    for p, t in zip(preds.ravel(), ts.ravel()):
+        if t == ignore_label:
+            continue
+        relevant[p] += 1
+        if p == t:
+            tp[p] += 1
+    return dtype(tp) / relevant
+
+
+def fbeta_score(precision, recall, beta=1.0):
+    beta_square = beta * beta
+    return ((1 + beta_square) * precision * recall /
+            (beta_square * precision + recall))
+
+
+def support(ts, dtype, label_num, ignore_label):
+    ret = numpy.zeros((label_num,), dtype=numpy.int32)
+    for t in ts.ravel():
+        if t == ignore_label:
+            continue
+        ret[t] += 1
+    return ret
+
+
+# Suppose label_num is 3 so that all valid label should be in [0, 3).\,
+# then, the typical output of this function is as follows:
+# has_ignore_label \ ignore_label | -1             | 0              |
+#       yes                       | 0,1,-1,2,-1... | 0,1,2,2,0,1... |
+#       no                        | 0,1,2,2,0,1... | 1,2,1,1,2,1... |
+def make_ground_truth(label_num, shape, ignore_label, has_ignore_label):
+    if (ignore_label == -1) != (has_ignore_label):  # xor
+        lower = 0
+    elif ignore_label == 0 and not has_ignore_label:
+        lower = 1
+    else:
+        lower = -1
+
+    t = numpy.random.randint(lower, label_num, shape)
+    return t.astype(numpy.int32)
+
+
+@testing.parameterize(
+    *testing.product_dict(
+        [{'y_shape': (100, 3), 't_shape': (100,)},
+         {'y_shape': (100, 3, 5), 't_shape': (100, 5)}],
+        [{'dtype': numpy.float16},
+         {'dtype': numpy.float32},
+         {'dtype': numpy.float64}],
+        [{'beta': 1.0},
+         {'beta': 2.0}],
+        [{'label_num': 3},
+         {'label_num': None}],
+        [{'ignore_label': -1},
+         {'ignore_label': 0}],
+        [{'has_ignore_label': True},
+         {'has_ignore_label': False}]
+    )
+)
+class TestClassificationSummary(unittest.TestCase):
+
+    def setUp(self):
+        t_upper = 3 if self.label_num is None else self.label_num
+        self.label_num = 3
+        self.y = numpy.random.uniform(-1, 1, self.y_shape).astype(self.dtype)
+        self.t = make_ground_truth(t_upper, self.t_shape,
+                                   self.ignore_label, self.has_ignore_label)
+        self.check_forward_options = {}
+        if self.dtype == numpy.float16:
+            self.check_forward_options = {'atol': 1e-4, 'rtol': 1e-3}
+
+        warnings.filterwarnings('ignore', category=RuntimeWarning)
```

Comment:
Could you add a comment that describe the reason why we suppress RuntimeWarning like that?
```python
        # Suppress warning that arises from zero division.
        warnings.filterwarnings('ignore', category=RuntimeWarning)
```